src/ffs/priv/node.ts

```diff
@@ -0,0 +1,110 @@
+import { emptyDir, encryptNode, decryptNode, decrypt, encrypt, contentToBytes, bytesToContent } from './helpers'
+import file from '../file'
+import { PrivateNodeData, Link, Node } from '../types'
+import { CID, FileContent } from '../../ipfs'
+
+export class PrivateNode implements Node {
+
+  links: Link[]
```

Comment:
Why not as a map?

Reply:
IPFS DAGNode's use an array and I was modeling our node after theirs. I don't think there's any reason that we need to stick to it though... I remember reading a discussion on this somewhere in IPFS's Github. I'll try and dig that up to see their rationale

Reply:
Yeah, let's make this work for our use cases. `Link[]` has issue with namespace collisions, &c. It's a level of detail that we don't probably don't want to deal with, but we can always convert back and forth (FFS -> DAGNode and DAGNode -> FFS)